sgtm/artemis/README.md

```diff
@@ -0,0 +1,273 @@
+
+#### Set up
+
+##### Auth
+
+If the server side container is deployed to App Engine or Cloud Run, then Google
+Tag Manager will use the service account attached to the instance for connecting
+to Firestore.
+
+If the server side container is deployed in a different Cloud provider to Google
+Cloud, please [follow these additional instructions](
+https://developers.google.com/tag-platform/tag-manager/server-side/manual-setup-guide#optional_include_google_cloud_credentials)
+to attach a Google Cloud service account to the deployment.
+
+This service account needs to have permission to access the Firestore data.
+
+1. Open the [IAM Service Accounts page](
+   https://console.cloud.google.com/iam-admin/serviceaccounts) in the Google
+   project that contains the sGTM container, and make a note of the service
+   account email.
+   ![Service account email](./img/auth-sgtm-service-account.png)
+2. Open the [IAM page](https://console.cloud.google.com/iam-admin/iam) for the
+   Firestore project, and press grant access.
+   ![Firestore project grant IAM access](./img/auth-firestore-iam.png)
+3. Add the service account email from step 1, and assign it the `Cloud Datastore
+   User` role ([docs](
+   https://cloud.google.com/iam/docs/understanding-roles#datastore-roles)).
+   ![IAM permissions](./img/auth-iam-permissions.png)
+
+
+##### Variables in Tag Manager
+
+1. Download the [Aretmis variable template](./../src/gtm/get_document_from_firestore.tpl) 
+   and [extraction template](./../src/gtm/extract_value_from_stringified_json_object.tpl) 
+   files. Ensure that the file extensions remain .tpl.
+2. Go to the server side container in
+   [tagmanager.google.com](https://tagmanager.google.com/).
+3. Go to templates -> new variable template.
+4. Click on the three-dot menu on the top right and choose `Import`.
+5. Select the "[Aretmis variable template](./../src/gtm/get_document_from_firestore.tpl)”
+    file.
+6. Go to the permission tab and set the permissions for Firestore, ensuring you
+   update the project ID.
+   ![Template permissions](./img/gtm-template-permissions.png)
+7. Save the template.
+8. Go to variables -> new user defined variable and create a “user” variable
+   from the Artemis variable template. You will need to pass in the use variable
+   that is being send from the client-side event which may required you creating 
+   a user ID variable depending on your set up.
+9. Go to templates -> new variable template.
+10. Click on the three-dot menu on the top right and choose `Import`.
+11. Select the [extraction template](./../src/gtm/extract_value_from_stringified_json_object.tpl) 
+    file.
+12. Save the template.
+13. Go to variables -> new user defined variable and create variables based on 
+    your (e.g. new/returning user, high_value_user) from the extraction variable
+    template.
+14. These variable are now available to be used in tags as metrics and dimensions
+11. Save and deploy the code.
+
+
+## Example Use-cases
```

Comment:
This bit is great! I'd put this far higher in the document, even before the background